src/Utilities/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataAbstractValueDomain.cs

```diff
@@ -0,0 +1,60 @@
+﻿
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+
+    internal partial class TaintedDataAnalysis
+    {
+        private sealed class TaintedDataAbstractValueDomain : AbstractValueDomain<TaintedDataAbstractValue>
+        {
+            public static readonly TaintedDataAbstractValueDomain Default = new TaintedDataAbstractValueDomain();
+
+            public override TaintedDataAbstractValue UnknownOrMayBeValue => TaintedDataAbstractValue.Unknown;
+
+            public override TaintedDataAbstractValue Bottom => TaintedDataAbstractValue.Unknown;
+
+            public override int Compare(TaintedDataAbstractValue oldValue, TaintedDataAbstractValue newValue)
+            {
+                // The newly computed abstract values for each basic block
+                // must be always greater or equal than the previous value
+                // to ensure termination.
+
+                // Unknown < NotTainted < Tainted
+                if (oldValue.Kind == TaintedDataAbstractValueKind.Tainted && newValue.Kind == TaintedDataAbstractValueKind.Tainted)
+                {
+                    return oldValue.SourceOrigins.Count.CompareTo(newValue.SourceOrigins.Count);
```

Comment:
>oldValue.SourceOrigins.Count.CompareTo(newValue.SourceOrigins.Count) [](start = 27, length = 68)

I would have expected a subset compare operation. Probably something like we do here: [https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Utilities/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeAbstractValueDomain.cs#L38](https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Utilities/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeAbstractValueDomain.cs#L38)

Reply:
Subset comparison makes more sense.  Will change it.